app/workers/scrape_job.rb

```diff
@@ -0,0 +1,97 @@
+class HardWorker
+  require 'capybara'
+  require 'capybara/poltergeist'
+
+  include Sidekiq::Worker
+  include Capybara::DSL
+  Capybara.register_driver :poltergeist do |app|
+    Capybara::Poltergeist::Driver.new(app, timeout: 60, js_errors: false)
+  end
+  Capybara.default_driver = :poltergeist
+
+  # perform scraping job through capybara/poltergeist
+  def perform(shop_name)
+
+    #set shop variable as ENV["shop"] as input when calling rake scraper:scrape
+    shop = shop_name
+
+    #visit the site
+    visit "http://salefinder.com.au/#{shop}-catalogue?qs=1,,0,0,0"
+
+    #set the location to Melbourne, 3000
+    page.find('a#header-change-region').trigger('click')
+    page.find('input#location-search').set("Melbourne, 3000")
+    page.find('div.autocomplete-suggestion').click
+
+    #getting total product count
+    product_count = page.find('span#product-count').text
+    product_count.slice!("Showing 0-12 of ")
+    product_count.slice!(" products for Woolworths")
+    product_count = product_count.to_i
+
+    #setting the max_page, first_page, and intermediate_page
+    max_page = (product_count.to_f/12).ceil
+    first_page = 1
+    page_number = 1
+    num_rescue = 0
+
+    begin
+      #loop from first_page to max_page
+      (page_number..max_page).each do |num|

```

Comment:
This works fine:
```ruby
(page_number..max_page).each do |num|
```
though the `page_number` variable is getting used for slightly different purposes at different times. For a little more clarity change the above line to:
```ruby
(first_page..max_page).each do |num|
```